doc/Type/Supply.pod6

```diff
@@ -370,6 +370,22 @@ Creates a "tail" supply with the same semantics as L<List.tail|/type/List#method
     my $ts = $s.tail(2);
     $ts.tap(&say);           # OUTPUT: «3␤2␤»
 
+=head2 method first
+
+    method first(Supply:D: :$end, |c)
+
+This method is a frontend on the C<grep> method. Its creates a new "head"
+supply, or a "tail" supply if the optional named parameter C<:end>
+is truthy, from a supply created by calling the C<grep> method on the
```

Comment:
Maybe you mean something like "bookend"? It returns either the first or the last element that meets that expression. Or ot emits the first of the remaining elements?